### ISSUE TITLE
Test Brave.com download button post ref code fix

### ISSUE THREAD

**Issue by AlanBreck** [brave/qa-resources#372]
#### Release Date/Target:

* Release Date/Completion Timing: ASAP?

#### Summary:
To date, ref codes have not been applied to macOS ARM64 downloads. This change both fixes several bugs with ref codes on referrer pages on the site, as well as allowing ref codes to be attached to downloads for `arm64`. A summary of requirements is below:

If no ref code in URL or localStorage, no ref code should be applied.
If ref code present in URL or found in localStorage, ref code should be applied to all download links, unless otherwise specified.
If on a specific referrer's page (e.g. brave.com/rogan, brave.com/annacramling, etc.), ref code from URL or localStorage **should not** be applied to download links on the page as those links should have the ref code for that referrer (e.g. `/rogan === WWK708`, `/anncramling === APX606`.
Ref codes should never be applied to non-release build channels (e.g. on brave.com/download-beta, etc.).

#### Related GH issues/Documentation:
https://github.com/brave/brave-com-resources/issues/232

**Comment by stephendonner** [comment]
Full changes along with commit, here: https://github.com/brave/brave-com/pull/416#issue-1008487301.

**Comment by AlanBreck** [comment]
Hi there! Just checking the status here. Thanks!

**Comment by stephendonner** [comment]
@AlanBreck Tried taking a crack at this today, and there are a few things I'm not clear on.

1. Can you give me a scenario with steps to get a ref code into localStorage, and how to verify that it's there?  I'm only familiar with them from limited client testing, where it sends the ref code on an ping, 2nd launch of Brave (and after 24 hours, I think?)
2. Can you help fill in/correct the following tests?
3. Are there other regression tests to add?
4. What are the "several bugs with refcodes on referrer pages", and how do I verify they're fixed?

### `arm64` ref codes

1. new profile
2. launch a browser on `arm64`
3. load `brave.com/download`
4. click on the `Download` button
5. ...
6. confirm 

### No referral codes

1. new profile
2. launch a browser
3. load `brave.com/download`
4. click on the `Download` button
5. ...
6. confirm there's no referral code

### Referrer's pages

1. new profile
2. launch a browser
3. load `brave.com/rogan` or `brave.com/annacramling`
4. click on the `Download` button
5. ...
6. confirm the referral codes are `WWK708` and `APX606`, respectively


### `brave.com/talk` referral code

1. new profile
2. launch a browser
3. load `brave.com/talk`
4. click on the `Download` button
5. ...
6. confirm the referral code is `QPU818`

### Non-release build channels

1. new profile
2. launch a browser
3. load the following sites:
   1. `https://brave.com/download-beta`
   2. `https://brave.com/download-nightly/`
4. click on the `Download` button
5. ...
6. confirm there's no referral code

**Comment by kjozwiak** [comment]
> Hi there! Just checking the status here. Thanks!

We'll get this prioritized this week. Right now, the current priority for QA was the following:

* Finally get `1.30.x` out (which happened last week)
* Release `1.30.x - Release #2` which updated chromium to `94.0.4606.71` which fixed a public UAF exploit
* run through C94 --> C95 upgrades so we can merge C95 into `master` via https://github.com/brave/brave-browser/issues/18533, https://github.com/brave/brave-browser/issues/18535, https://github.com/brave/brave-browser/issues/18536 & https://github.com/brave/brave-browser/issues/18537.

**Comment by stephendonner** [comment]
I added and tested Windows and macOS tests; will probably need to add Android and Linux, here, too:

https://github.com/brave/qa-resources/wiki/Brave.com-referrer-code-tests

**Comment by AlanBreck** [comment]
FWIW: Linux doesn't get ref codes.

**Comment by stephendonner** [comment]
> FWIW: Linux doesn't get ref codes.

Yup, I meant a regression test (that there are no changes/refcodes) 👍

**Comment by AlanBreck** [comment]
On the `brave.com/talk` test, it's actually non-Brave browsers that should see the download CTA; Brave users should see "start free call." Can you confirm?
